### PR TITLE
Changing the role of GroupedList from "grid" to "group" 

### LINKED ISSUE #1672: [GroupedList] role=grid but no rowgroup or row underneath in the dom
The GroupedList control has a role of "grid" but there are no rows or rowgroups roles in its child elements. It looks like its actually a list underneath with no grid implementation. 
https://www.w3.org/TR/wai-aria/roles#grid

### PR DESCRIPTION
**Pull request by aadhars**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1672 

#### Description of changes

Changed the role of GroupedList from "grid" to "group". If we see GroupedList is designed  as composition of  list and listitems. From w3c standard guideline if we assign it a role as grid we should have rowgroups and row inside it, which is not the case.

Role group can contain list and listitem. Also a element withe role "group" can have nested childrens with role="group" 

List can also contains groups which further can contain more list , groups and listitems.

More info about roles can be seen here: 

https://www.w3.org/TR/wai-aria/roles#list
https://www.w3.org/TR/wai-aria/roles#group

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class GroupedList extends BaseComponent<IGroupedListProps, IGroupedListSt
         className={ css('ms-GroupedList', styles.root, className) }
         data-automationid='GroupedList'
         data-is-scrollable='false'
-        role='grid'
+        role='group'
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.tsx]: should we use `rowgroup` then ? https://www.w3.org/TR/wai-aria/roles#rowgroup

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx [hunk 2]
```diff
@@ -191,6 +191,7 @@ export class GroupedListSection extends BaseComponent<IGroupedListSectionProps,
       <div
         ref='root'
         className={ css('ms-GroupedList-group', styles.group, this._getDroppingClassName()) }
+        role='group'
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx]: same here, should we be specific and use ` rowgroup` instead?

Comment (aadhars) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedListSection.tsx]: Rowgroup required its child element to have atleast one role="row". Internally we have designed groupedList as List with ListItem and not as row

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.tsx (hunk 1 above)
Comment (aadhars) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupedList.tsx]: Rowgroup requires its child element to have atleast one role="row".
Required Owned Elements:	row
Internally we have designed groupedList as List with ListItem and not as row

**Comment by betrue-final-final** [comment]
@cschlecty, is this correct?

**Comment by dzearing** [comment]
@ThomasMichon may want to look here.

**Comment by aadhars** [comment]
@ThomasMichon Can you have a look at this PR

**Comment by aadhars** [comment]
@dzearing @micahgodbolt Can you help getting this reviewed

**Comment by micahgodbolt** [comment]
I pinged Thomas. We need to verify that this is the correct solution before pulling it in.

**Comment by betrue-final-final** [comment]
Yes @micahgodbolt, I'm not quite sure about this.

**Comment by ThomasMichon** [comment]
Hey, looking at this, I think the right role for `GroupedList` is simply `presentation`. You need the accessibility tree to only see the `role='list'` and `role='listitem'` elements and nothing else. So anything in between should simple be marked as `presentation`.

**Comment by aadhars** [comment]
Got it @ThomasMichon . I have changed the role to presentation

**Comment by aadhars** [comment]
@ThomasMichon Can you have a look at the changes.

**Comment by dzearing** [comment]
pinging on this today, we'll get this reviewed.

**Comment by aadhars** [comment]
@dzearing @ThomasMichon Thanks for getting back on this. Can anyone of you complete the pull request

**Comment by jspurlin** [comment]
I agree that role="grid" needs to be removed, but this component needs more work from an accessibility standpoint. A GroupedList is really a tree and should be marked up (and behave) as such.

I've opened [GroupedList should be represented as a [tree](https://www.w3.org/TR/wai-aria/roles#tree)/[treeitem](https://www.w3.org/TR/wai-aria/roles#treeitem) (as well as the keyboard navigation) #2084](https://github.com/OfficeDev/office-ui-fabric-react/issues/2084) to track that work.

See: https://www.w3.org/TR/2009/WD-wai-aria-practices-20090224/#TreeView for a description of the expected behavior

**Pull request merged**